core/cog-viewport.c

```diff
@@ -165,6 +165,9 @@ cog_viewport_dispose(GObject *object)
 static void
 cog_viewport_finalize(GObject *object)
 {
+    // Unset the reference to this viewport in the attached views.
+    cog_viewport_foreach((CogViewport *) object, (GFunc) cog_view_set_viewport, NULL);
+
```

Comment:
We don't need this if we make the views keep a weak reference to their current viewport, as per my suggestion above.

Reply:
I agree. A weak references works much better and it is a simpler solution. Latests changes goes in this direction. Thanks for the suggestion.